File: seaplane-sdk/python/src/seaplane/model/provider.py
from enum import Enum
from typing import List, Optional


class Provider(Enum):
    aws = "AWS"
    azure = "AZURE"
    digital_ocean = "DIGITALOCEAN"
    equinix = "EQUINIX"
    gcp = "GCP"

    def __str__(self) -> str:
        return str(self.value)


def to_providers(providers: Optional[List[str]]) -> List[Provider]:
    if not providers:
        return []

    return [Provider(provider) for provider in providers]

File: seaplane-sdk/javascript/src/model/provider.ts
export enum Provider {
  AWS = "AWS",
  Azure = "AZURE",
  DigitalOcean = "DIGITALOCEAN",
  Equinix = "EQUINIX",
  GCP = "GCP"
}

export const mapToProvider = (providers?: string[]): Provider[]  => {
  if (!providers) {
    return []
  }

  return providers.map(provider => Provider[provider])
}